lib/vanagon/platform.rb

```diff
@@ -435,6 +435,25 @@ def add_build_repository(*args)
       end
     end
 
+    def generate_compiled_archive(project)
+      metadata = project.build_manifest_json(true)
+      metadata.gsub!(/\n/, '\n')
+      target_dir = project.repo ? output_dir(project.repo) : output_dir
+      name_and_version = "#{project.name}-#{project.version}"
+      archive_dir = "#{project.name}-archive"
```

Comment:
Minor (feel free to ignore): 

* I like spelling out names, archive_directory. I think it makes code easier to read. 
* Unless there are portability concerns, using the --long-options to commands makes them clearer

Reply:
Unfortunately for this there are portability concerns (as this will run on all the platforms we support) so I've got to go with the short options.

I can fix at least some of the `dir` -> `directory`, but I think that shortening is pretty pervasive throughout the codebase so some of that might not make sense to do here.

Reply:
I'd be happy with consistency inside of new methods.